Lib/sqlite3/__init__.py

```diff
@@ -55,3 +55,16 @@
 """
 
 from sqlite3.dbapi2 import *
+from sqlite3.dbapi2 import (_deprecated_names,
+                            _deprecated_version_info,
+                            _deprecated_version)
+
+
+def __getattr__(name):
+    if name in _deprecated_names:
+        from warnings import warn
+
+        warn(f"{name} is deprecated and will be removed in Python 3.14",
+             DeprecationWarning, stacklevel=2)
+        return globals()[f"_deprecated_{name}"]
+    raise AttributeError(f"module {__name__!r} has no attribute {name!r}")
```

Comment:
Isn't it enough to emit the warning in dbapi2.py? Do we need it here as well?

Reply:
In most of the doc's, references to version or version_info are directly through `sqlite3` and not `sqlite3.dbapi2`, I thought it might be good to also emit the warning in __init__.py as well. Do you want me to remove it from `__init__.py`?

Reply:
We need both, otherwise the deprecation warning wouldn't be emitted on `sqlite3.version`. Importing as e..g. `from sqlite3.dbapi2 import _deprecated_version_info as version_info` doesn't trigger the deprecation warning on attribute access (or at all, it seems).

Simple test script:

```python
import sqlite3, sqlite3.dbapi2, warnings
warnings.simplefilter("always")
sqlite3.version
sqlite3.version_info
sqlite3.dbapi2.version
sqlite3.dbapi2.version_info
```

A

Reply:
Right, that answers my question. (I did not have access to my computer to try it out when I commented earlier today.)